core/field.js

```diff
@@ -475,15 +474,24 @@ Blockly.Field.prototype.updateColour = function() {
  */
 Blockly.Field.prototype.render_ = function() {
   this.textElement_.textContent = this.getDisplayText_();
-  this.updateWidth();
+  this.updateSize_();
 };
 
 /**
- * Updates the width of the field. This calls getCachedWidth which won't cache
- * the approximated width on IE/Edge when `getComputedTextLength` fails. Once
- * it eventually does succeed, the result will be cached.
+ * Updates the width of the field. Redirects to updateSize_().
+ * @deprecated May 2019 Use Blockly.Field.getCachedWidth() directly instead.
```

Comment:
Why does this say to use `getCachedWidth` instead of `updateSize_`?

Reply:
render_ only supports text fields. updateSize_ is only called by the default render_ function. updateSize_ also only supports text fields.

If you have a text field, there is no need to override render_, and no need to call updateSize_.

If you have an image, or mixed field you need to override render_. In this case updateSize_ will not be helpful, because it only supports text fields. But if it is a mixed field, you may still want to know the size of your text, so you should call getCachedWidth().

Reply:
I see your point, but the redirect still feels wrong to me:
- `updateWidth` and `updateSize` have no return value, but have side effects
- `getCachedWidths` has a return value and no side effects

I suggest:
@deprecated May 2019 Use Blockly.Field.updateSize_() to force an update to the size of the field, or Blockly.Field.getCachedWidth() to check the size of the field..

Reply:
Sounds good, Fixed.